### PR TITLE
Freezie improvements

### PR DESCRIPTION
**Pull request by Linkaaaaa**
# Freezie
## Buffs
- Giant Snowball Target
- Aurora Beam Target
## Mechanics
- Targetted by Giant Snowball
- Targetted by Aurora Beam
- Hit by Aurora Beam
- Hit by Giant Snowball
- Hit by Blizzard
- Hit by Frost Patch
- Hit by Jutting Ice Spike
- Casted Fire Snowball
- Enemy received icy Barrier
- Enemy casted Aurora beam
- Enemy casted Frost Patch
- Enemy casted Giant Snowball
## Instant Casts
- Cold Hearted Aura
## Agents
- Converted snow piles gadget to agents
## Decorations
- Fixation tether from Icy Protector to Freezie
- Fixation icon for Aurora Beam
- Fixation icon for Giant Snowball
- Frozen Patch
- Giant Snowball AoE and similar (same effect)
- Blizzard
## Damage Modifier
- Negative damage modifier for Icy Barrier

# General
- Fixed tagGUID for older logs, was triggering an exception because it was using a string containing the icon url instead of the GUID.

# TODO
- Add incoming damage modifier for his enrage (once supported)
- Missing combat replay map image and size (waiting on Linus :D)
- Couldn't figure out a way to display Aurora Beam in the decorations because it follows a player

### CODE REVIEW

File: GW2EIEvtcParser/EIData/CombatReplay/CombatReplay.cs [hunk 1]
```diff
@@ -1,6 +1,7 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using GW2EIEvtcParser.EncounterLogic;
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EIData/CombatReplay/CombatReplay.cs:4]: ?

File: GW2EIEvtcParser/EIData/DamageModifiers/CommonDamageModifiers/EncounterDamageModifiers.cs [hunk 2]
```diff
@@ -67,6 +67,7 @@ internal static class EncounterDamageModifiers
                     Segment segment = target.GetBuffStatus(log, IonShield, ahde.Time);
                     return segment.Value >= 20;
                 }),
+            new BuffOnFoeDamageModifier(IcyBarrier, "Icy Barrier", "-10% per stack", DamageSource.All, -10.0, DamageType.StrikeAndCondition, DamageType.All, Source.FightSpecific, ByStack, BuffImages.ShieldOfIce, DamageModifierMode.PvE),
```
Comment (EliphasNUIT) [inline GW2EIEvtcParser/EIData/DamageModifiers/CommonDamageModifiers/EncounterDamageModifiers.cs:70]: This combines with vulnerability and it is also possible for it to reach 100% reduction. Check out how that is managed on PowerOfTheVoid, DagdaDemonicAura, PillarPandemonium

File: GW2EIEvtcParser/EIData/CombatReplay/CombatReplay.cs (hunk 1 above)
Comment (Linkaaaaa) [reply GW2EIEvtcParser/EIData/CombatReplay/CombatReplay.cs:4]: I was checking something out and visual studio added it without me noticing, I'll remove it

**Comment by Linkaaaaa** [comment]
I've commented out the combat replay map section for now, linus is busy until mid next week

**Comment by EliphasNUIT** [comment]
Thanks!

**Pull request merged**